src/settings.ts

```diff
@@ -144,22 +170,95 @@ export class FullCalendarSettingTab extends PluginSettingTab {
 			async (source: CalendarSource) => {
 				sourceList.addSource(source);
 			},
-			() => sourceList.state.sources
-				.map((s) => s.type === "local" && s.directory)
-				.filter((s): s is string => !!s)
+			() =>
+				sourceList.state.sources
+					.map((s) => s.type === "local" && s.directory)
+					.filter((s): s is string => !!s)
 		);
 
 		const sourcesDiv = containerEl.createDiv();
 		sourcesDiv.style.display = "block";
 		let sourceList = ReactDOM.render(
 			createElement(CalendarSettings, {
 				sources: this.plugin.settings.calendarSources,
-				submit: async (settings: CalendarSource[]) => {
-					this.plugin.settings.calendarSources = settings;
+				submit: async (sources: CalendarSource[]) => {
+					await this.handleRemoval(
+						this.plugin.settings.calendarSources,
+						sources
+					);
+
+					this.plugin.settings.calendarSources = sources;
 					await this.plugin.saveSettings();
 				},
+				sourcesUpdated: async (
+					oldSources: CalendarSource[],
+					sources: CalendarSource[]
+				) => {
+					await this.handleRemoval(oldSources, sources, true);
+				},
 			}),
 			sourcesDiv
 		);
+		this.sourceList = sourceList;
+	}
+
+	async handleRemoval(
```

Comment:
```suggestion
	async handleCalendarSourceRemoval(
```

Just to make the name more specific.